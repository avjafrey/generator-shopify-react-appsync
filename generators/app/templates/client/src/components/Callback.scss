.main {
    margin-top: 100px;
}

.header {
    text-align: center;
    margin-bottom: 50px;
}

.body {
    width: 400px;
    margin: auto;
}