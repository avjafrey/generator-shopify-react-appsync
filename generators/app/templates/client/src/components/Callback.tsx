import { Banner, DisplayText } from "@shopify/polaris";
import * as React from "react";

import { Spinner } from "./Spinner";

import * as styles from "./Callback.scss";

interface ICallbackProps {
    errorMessage: string | null;
    loginUrl: string;
}

export function Callback(props: ICallbackProps) {
    const body = props.errorMessage ? (
        <Banner
            title="Authentication Error"
            status="critical"
            action={{ content: "Try again", url: props.loginUrl }}
        >
            <p>{props.errorMessage}</p>
        </Banner>
    ) : (
            <div>
                <Spinner />
                <p>Redirecting, please wait...</p>
            </div>
        );
    return (
        <main role="main" className={styles.main}>
            <header className={styles.header}>
                <DisplayText size="extraLarge">Shopify App â€” Installation</DisplayText>
            </header>
            <div className={styles.body}>
                {body}
            </div>
        </main>
    );
}
