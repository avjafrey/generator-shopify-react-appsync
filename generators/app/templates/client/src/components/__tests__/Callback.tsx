import { AppProvider } from "@shopify/polaris";
import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { Callback } from "../Callback";

test("Renders correctly with no error", () => {
  const login = (
    <AppProvider>
      <Callback errorMessage={null} loginUrl="/" />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with error", () => {
  const login = (
    <AppProvider>
      <Callback errorMessage={"Something went wrong"} loginUrl="/" />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
