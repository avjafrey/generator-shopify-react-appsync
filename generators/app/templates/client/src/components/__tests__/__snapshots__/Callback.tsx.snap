// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with error 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <div
      aria-describedby="Banner1Content"
      aria-labelledby="Banner1Heading"
      aria-live="polite"
      className="Polaris-Banner Polaris-Banner--statusCritical"
      onMouseUp={[Function]}
      role="alert"
      tabIndex={0}
    >
      <div
        className="Polaris-Banner__Ribbon"
      >
        <span
          className="Polaris-Icon Polaris-Icon--colorRedDark Polaris-Icon--isColored Polaris-Icon--hasBackdrop"
        >
          <svg
            aria-hidden="true"
            className="Polaris-Icon__Svg"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g  fill-rule=\\"evenodd\\"><circle fill=\\"currentColor\\" cx=\\"10\\" cy=\\"10\\" r=\\"9\\"/><path d=\\"M2 10c0-1.846.635-3.543 1.688-4.897l11.209 11.209A7.954 7.954 0 0 1 10 18c-4.411 0-8-3.589-8-8m14.312 4.897L5.103 3.688A7.954 7.954 0 0 1 10 2c4.411 0 8 3.589 8 8a7.952 7.952 0 0 1-1.688 4.897M0 10c0 5.514 4.486 10 10 10s10-4.486 10-10S15.514 0 10 0 0 4.486 0 10\\" /></g>",
              }
            }
            focusable="false"
            viewBox="0 0 20 20"
          />
        </span>
      </div>
      <div>
        <div
          className="Polaris-Banner__Heading"
          id="Banner1Heading"
        >
          <p
            className="Polaris-Heading"
          >
            Authentication Error
          </p>
        </div>
        <div
          className="Polaris-Banner__Content"
          id="Banner1Content"
        >
          <p>
            Something went wrong
          </p>
          <div
            className="Polaris-Banner__Actions"
          >
            <div
              className="Polaris-ButtonGroup"
            >
              <div
                className="Polaris-ButtonGroup__Item"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <a
                  className="Polaris-Button Polaris-Button--outline"
                  data-polaris-unstyled={true}
                  href="/"
                  onMouseUp={[Function]}
                >
                  <span
                    className="Polaris-Button__Content"
                  >
                    <span>
                      Try again
                    </span>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`Renders correctly with no error 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <div>
      <div>
        <svg
          className="Polaris-Spinner Polaris-Spinner--colorTeal Polaris-Spinner--sizeLarge"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M15.542 1.487A21.507 21.507 0 0 0 .5 22c0 11.874 9.626 21.5 21.5 21.5 9.847 0 18.364-6.675 20.809-16.072a1.5 1.5 0 0 0-2.904-.756C37.803 34.755 30.473 40.5 22 40.5 11.783 40.5 3.5 32.217 3.5 22c0-8.137 5.3-15.247 12.942-17.65a1.5 1.5 0 1 0-.9-2.863z\\" />",
            }
          }
          role="status"
          viewBox="0 0 44 44"
        />
      </div>
      <p>
        Redirecting, please wait...
      </p>
    </div>
  </div>
</main>
`;
